nni/experiment/config/remote.py

```diff
@@ -0,0 +1,52 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from dataclasses import dataclass
+from pathlib import Path
+from typing import List, Optional, Union
+
+from .base import ConfigBase, PathLike
+from .common import TrainingServiceConfig
+from . import util
+
+__all__ = ['RemoteConfig', 'RemoteMachineConfig']
+
+
+@dataclass(init=False)
+class RemoteMachineConfig(ConfigBase):
+    host: str
+    port: int = 22
+    user: str
+    password: Optional[str] = None
+    ssh_key_file: PathLike = '~/.ssh/id_rsa'
```

Comment:
The ssh_key is optional key, is used to connect remote machine, not local file. I think set it to None as default is better.

Reply:
As said in doc, the behavior is:
  - when password is explicitly set, use password
  - when password is not set and identity file is explicitly set, use specified identify file
  - when both are not set, use default identity file

I expect this behavior because when using openssh directly, I don't need to specify anything when I put identify file at `~/.ssh/id_rsa`.
I know it can be other algorithms like `id_edXXX` though, but I believe they are rarely used.